src/python/test_LossFunctionTranslation.py

```diff
@@ -1,128 +1,156 @@
 import unittest
 import json
 from vehicle import LossFunctionTranslation
+from vehicle import generate_loss_function
 from tensorflow import keras
+import tensorflow as tf
+import numpy as np
 
 class TestLossFunctionTranslation(unittest.TestCase):
     def load_json(self, file_name):
         path_to_json = f'./src/python/test_json/{file_name}.json'
         with open(path_to_json) as f:
             json_dict = json.load(f)
         return json_dict
+    
+    def vcl_file(self, file_name):
+        path_to_vcl = f'./src/python/test_json/{file_name}.vcl'
+        return path_to_vcl
 
     def test_constant(self):
+        path_to_vcl = self.vcl_file('test_constant')
+        functionName = 'constant'
         resources = {}
-        json_dict = self.load_json('test_constant')
-        context = []
-        loss = LossFunctionTranslation().to_loss_function(resources, json_dict)(context)
-        self.assertEqual(loss, 0)
-
+        loss = generate_loss_function(path_to_vcl, functionName, resources)
+        self.assertEqual(loss(), 5)
     
-    def test_variable(self):
-        resources = {}
-        json_dict = self.load_json('test_variable')
-        context = [5, 2]
-        loss = LossFunctionTranslation().to_loss_function(resources, json_dict)(context)
-        self.assertEqual(loss, 2)
+    
+    # def test_variable(self):
+    #     path_to_vcl = self.vcl_file('test_variable')
+    #     functionName = 'variable'
+    #     resources = {}
+    #     loss = generate_loss_function(path_to_vcl, functionName, resources)
+    #     self.assertEqual(loss(2), 2)
     
 
     def test_tensor(self):
+        path_to_vcl = self.vcl_file('test_tensor')
+        functionName = 'tensor'
         resources = {}
-        json_dict = self.load_json('test_tensor')
-        context = [5, 2]
-        loss = LossFunctionTranslation().to_loss_function(resources, json_dict)(context)
-        self.assertEqual(loss.shape, (1,))
-        self.assertEqual(loss, 5)
+        loss = generate_loss_function(path_to_vcl, functionName, resources)
+        self.assertEqual((np.array(loss().shape) & np.array(tf.constant([2, 4, 1, 0]).shape)).all(), True)
+        self.assertEqual((np.array(loss()) & np.array([5, 2, 16, 7])).all(), True)
```

Comment:
What does the `&` operator do here? Why do we need to cast `loss()` to `np.array`?

Reply:
The end operator and the np cast are needed because tensors cannot be compared. This is the only way I have found to compare the values and the shape.

Reply:
Could you possibly extract that into a function with a name that conveys what you're doing here?

Reply:
Yup, something like `def assertTensorsEqual(t1, t2):`